hack/tools/tilt-prepare/main.go

```diff
@@ -165,6 +172,30 @@ func readTiltSettings(path string) (*tiltSettings, error) {
 	return ts, nil
 }
 
+func checkproviderFileFormat(path string) (string, bool) {
+	if _, err := os.Stat(path + "/tilt-provider.yaml"); err == nil {
+		return path + "/tilt-provider.yaml", true
+	}
+	if _, err := os.Stat(path + "/tilt-provider.json"); err == nil {
+		return path + "/tilt-provider.json", true
+	}
+	return "", false
```

Comment:
Let's return an error instead of false here. Then you can just error check one level above and return the error if there is one